splight_lib/client/datalake/remote_client.py

```diff
@@ -262,3 +266,74 @@ def execute_query(
         df.drop(labels="Unnamed: 0", axis=1, inplace=True)
         df.set_index("timestamp", inplace=True)
         return df
+
+
+class BufferedRemoteDatalakeClient(RemoteDatalakeClient):
+    _PREFIX = "v2/engine/datalake"
+
+    def __init__(
+        self,
+        base_url: str,
+        access_id: str,
+        secret_key: str,
+        buffer_size: int = 10,
+        buffer_timeout: float = 30,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(
+            base_url,
+            access_id,
+            secret_key,
+            buffer_size,
+            buffer_timeout,
+            *args,
+            **kwargs,
+        )
+        self._base_url = furl(base_url)
+        token = SplightAuthToken(
+            access_key=access_id,
+            secret_key=secret_key,
+        )
+        self._restclient = SplightRestClient()
+        self._restclient.update_headers(token.header)
+
+        self._data_buffers = {
+            "default": DatalakeDocumentBuffer(buffer_size, buffer_timeout),
+            "routine_evaluations": DatalakeDocumentBuffer(
+                buffer_size, buffer_timeout
+            ),
+        }
+        self._flush_thread = Thread(target=self._flusher, daemon=True)
+        self._flush_thread.start()
+        logger.debug(
+            "Buffered Remote datalake client initialized.",
+            tags=LogTags.DATALAKE,
+        )
+
+    def save(
+        self, collection: str, instances: Union[List[Dict], Dict]
+    ) -> List[Dict]:
+        instances = instances if isinstance(instances, List) else [instances]
+        if collection not in self._data_buffers:
+            raise Exception("Invalid collection name")
+        self._data_buffers[collection].add_documents(instances)
```

Comment:
At some point shouldn't we have to save the data to the datalake too?

Reply:
There is a thread that is responsible for sending the documents to the datalake

Reply:
Oh yeah that's true!